### PR TITLE
Improved surface anchors

### PR DESCRIPTION
**Pull request by benptc**
Works in AR using groundplane for raycasts if gltf model isn't available. Moves tools in realtime when dragging the anchor instead of waiting for pointerup event, using a subtle tween animation. 

spatial-sensor-anchor-ios

### CODE REVIEW

File: src/gui/ar/groundPlaneAnchors.js [hunk 1]
```diff
@@ -21,13 +28,20 @@ createNameSpace("realityEditor.gui.ar.groundPlaneAnchors");
...
     let originColor = 0xffffff;
     let xBoxColor = 0xff0000;
     let zBoxColor = 0x0000ff;
     let selectionColor = 0xffff00;
-    let mouseCursorColor = 0xff00ff;
+    let mouseCursorColor = 0xffffff;
     let mouseCursorMesh = null;
+    let initialCalculationMesh = null;
+
+    const REALTIME_DRAG_UPDATE = true; // TODO: this looks a bit jittery for now, turn on if we improve its performance
```
Comment (dangond-ptc) [inline src/gui/ar/groundPlaneAnchors.js:42]: Is this meant to be on or off? The comment seems to imply off.

File: src/gui/ar/groundPlaneAnchors.js [hunk 2]
```diff
@@ -180,7 +227,7 @@ createNameSpace("realityEditor.gui.ar.groundPlaneAnchors");
             touchEventCatcher.style.top = '0';
             touchEventCatcher.style.width = '100vw';
             touchEventCatcher.style.height = '100vh';
-            let zIndex = 8000;
+            let zIndex = 2900;
             touchEventCatcher.style.zIndex = zIndex + 'px';
```
Comment (dangond-ptc) [inline src/gui/ar/groundPlaneAnchors.js:231]: Not from your changes, but zIndex should be set as a unitless property.

**Review by dangond-ptc — CHANGES_REQUESTED** [review]

**Review by hobinjk-ptc — APPROVED** [review]
LGTM with the changes Dan suggested

File: src/gui/ar/groundPlaneAnchors.js (hunk 1 above)
Comment (benptc) [reply src/gui/ar/groundPlaneAnchors.js:42]: Oh the comment is out of date, will fix – thanks!

**Review by dangond-ptc — APPROVED** [review]

**Pull request merged**